python/cog/server/worker.py

```diff
@@ -172,7 +173,11 @@ def run(self) -> None:
         self._stream_redirector.start()
 
         self._setup()
-        self._loop()
+        try:
+            loop = asyncio.get_running_loop()
+        except RuntimeError:
+            loop = asyncio.new_event_loop()
+        loop.run_until_complete(self._loop())
```

Comment:
What's the reasoning behind this `try/except` vs calling `asyncio.run(self._loop())`?

Reply:
I wasn't sure if there would be an event loop already running but I think you're right and this is the top of the subprocess so unless multiprocesisng does something I don't understand that should be fine